pkg/proxy/userspace/proxier.go

```diff
@@ -509,6 +514,7 @@ func (proxier *Proxier) mergeService(service *v1.Service) sets.String {
 			if err := proxier.cleanupPortalAndProxy(serviceName, info); err != nil {
 				klog.Error(err)
 			}
+			info.finished.Done()
```

Comment:
can this be called more than once (sending `finished` negative)? if we `continue` above on line 510, do we miss calling this when we should?

Reply:
@liggitt I don't think 510 is an issue, because mergeService() is trying to update the proxy if the service changed. Line 510 means nothing changed so leave it as-is. If the service did change, and it's already in the map, it'll be removed (cleanupPortalAndProxy()) and then re-added with a new ServiceInfo/waitgroup by addServiceOnPortInternal()